src/components/views/elements/OverlaySpinner.js

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2020 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, {useEffect, useRef, useState} from 'react';
+import Spinner from "./Spinner";
+
+/*
+ * A component which measures its children, and places a spinner over them
+ * while 'active' is true.
+ * We use 'holdOff' and 'holdOver' to shift the animations in time:
+ * We don't animate instantly to avoid a quick 'ghost' spinner for fast round trips.
+ * We have a short holdOver to allow the fade out to occur.
+ */
+
+export default function OverlaySpinner({active, className, children}) {
+    const measured = useRef(null);
+    const [state, setState] = useState({w: 0, h: 0});
+
+    const firstMount = useRef(true);
+    const [holdOver, setHoldOver] = useState(false);
+    const [holdOff, setHoldOff] = useState(false);
+
+    /* Follow the size of the element we're meant to cover */
+    useEffect(() => {
+        const interval = requestAnimationFrame(() => {
+            if (!measured.current) return;
+            setState({
+                w: measured.current.clientWidth,
+                h: measured.current.clientHeight,
+            });
+        });
+        return () => cancelAnimationFrame(interval);
+    });
+
+    /* If it's not the first mount and the state changes to inactive,
+     * set 'holdOver' to true so the exit animation can play */
+    useEffect(() => {
+        if (firstMount.current) {
+            firstMount.current = false;
+            return;
+        }
+        if (!active) {
+            const handle = setTimeout(setHoldOver, 200, false);
+            setHoldOver(true);
+            return () => { setHoldOver(false); clearTimeout(handle); };
+        }
+        if (active) {
+            const handle = setTimeout(setHoldOff, 200, false);
+            setHoldOff(true);
+            return () => { setHoldOff(false); clearTimeout(handle); };
+        }
+    }, [active]);
+
+    const visibility = !holdOff && active ? "visible" : "hidden";
+
+    return (<div className={className} style={{ position: "relative" }}>
```

Comment:
this style should get put into the css

Reply:
I disagree: this is the necessary and sufficient condition to make the absolute positioning of the subcomponent work –– I don't think anything is gained by having a CSS class to manipulate here, whereas the downside is that the `absolute`/`relative` pair is moved into another file where it's less easy to see what's going on

Reply:
I guess its just a pattern of our codebase that we try and avoid inline styles where possible.